File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        st = deque()
        st.append(root)
        res = []
        
        while st:
            l = len(st)
            level = []
            for i in range(l):
                node = st.popleft()
                if node:
                    level.append(node.val)
                    st.append(node.left)
                    st.append(node.right)
            if level:
                res.append(level)
        
        return res

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {

        List<List<Integer>> res = new ArrayList<>();
        if (root == null) return res;
        
        Queue <TreeNode> st = new LinkedList<>();
        st.add(root);
        
        while (!st.isEmpty()){
            int length = st.size();
            List<Integer> level = new ArrayList<>();
            
            for (int i = 0; i < length; i++){
                TreeNode node = st.poll();
                level.add(node.val);
                if (node.left != null){    
                    st.add(node.left);
                }
                if (node.right != null){    
                    st.add(node.right);
                }
            }
            res.add(level);
        }
        
        return res;
    }
}